compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -3337,8 +3337,14 @@ boolean validateErrorVariable(BLangErrorVariable errorVariable, SymbolEnv env) {
                 defineMemberNode(errorVariable.message, env, symTable.stringType);
             }
 
-            BLangVariable cause = errorVariable.cause;
-            if (cause != null) {
+            BLangVariable errorCause = errorVariable.cause;
+            if (errorCause != null) {
+                if (errorCause.getKind() == NodeKind.VARIABLE &&
+                        names.fromIdNode(((BLangSimpleVariable) errorCause).name) == Names.IGNORE) {
+                    dlog.error(errorCause.pos,
+                            DiagnosticErrorCode.CANNOT_USE_WILDCARD_BINDING_PATTERN_FOR_ERROR_CAUSE);
```

Comment:
When considering the following example, is it correct to give this error?
```ballerina
type SampleErrorData record {|
    int code;
    string reason;
|};

type SampleError error<SampleErrorData>;

public function main() {
    _ = ();
    SampleError sampleErr = error("Transaction Failure", (), code = 20,
                            reason = "deadlock condition");

    var error(_, _, code = code, reason = reason) = sampleErr;
}


```

Reply:
Can you please elaborate on why you think it shouldn't be an error?

Reply:
The value belongs to type `()`. Therefore, why can't we use a wildcard pattern with it

Reply:
```ballerina
type SampleErrorData record {|
    int code;
    string reason;
|};

type SampleError error<SampleErrorData>;

public function main() {
    SampleError sampleErr = error("Transaction Failure", "error", code = 20,
                            reason = "deadlock condition");

    var error(message, _, code = code, reason = reason) = sampleErr;
}
```
In this example, we get an error that says 'string' cannot be assigned to 'error?'. And also it gives that wildcard binding patterns can only be used with a value that belongs to the 'any' type. My question is, why can't we use a wildcard pattern if we are using a string value?

Reply:
We don't do that level of data flow analysis, do we? What we type check is for the generic scenario.

```ballerina
function fn(SampleError sampleErr) {
    var error(message, _, code = code, reason = reason) = sampleErr;
}
```

We don't know if a cause will always be specified or not specified, so we have to analyze for all the possibilities, which means it could be error.